File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int res = 0;

        for (int num : nums) {
            if (!map.containsKey(num)) {
                map.put(num, 1 + map.getOrDefault(num - 1, 0) + map.getOrDefault(num + 1, 0));
                map.put(num - map.getOrDefault(num - 1, 0), map.get(num));
                map.put(num + map.getOrDefault(num + 1, 0), map.get(num));
                res = Math.max(res, map.get(num));
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-5.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod = 1
        zeroes = 0
        res = []

        for num in nums:
            if num == 0:
                zeroes += 1
                if zeroes >= 2:
                    prod *= num
            else:
                prod *= num

        for num in nums:
            if zeroes >= 2:
                res.append(0)
            elif zeroes == 1:
                if num == 0:
                    res.append(prod)
                else:
                    res.append(0)
            else:
                res.append(int(prod / num))
        return res